sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIOStorageWriteIT.java

```diff
@@ -0,0 +1,131 @@
+import com.google.common.collect.ImmutableList;
+import java.io.IOException;
+import org.apache.beam.sdk.Pipeline;
+import org.apache.beam.sdk.extensions.gcp.options.GcpOptions;
+import org.apache.beam.sdk.io.GenerateSequence;
+import org.apache.beam.sdk.io.gcp.testing.BigqueryClient;
+import org.apache.beam.sdk.options.PipelineOptionsFactory;
+import org.apache.beam.sdk.testing.TestPipeline;
+import org.apache.beam.sdk.transforms.DoFn;
+import org.apache.beam.sdk.transforms.ParDo;
+import org.apache.beam.sdk.transforms.SerializableFunction;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/**
+ * Integration tests for {@link
+ * org.apache.beam.sdk.io.gcp.bigquery.BigQueryIO#write(SerializableFunction)}. This test writes
+ * 30MB data to BQ and verify the written row count.
+ */
+@RunWith(JUnit4.class)
+public class BigQueryIOStorageWriteIT {
+
+  private enum WriteMode {
+    EXACT_ONCE,
+    AT_LEAST_ONCE
+  };
+
+  private String project;
+  private static final String DATASET_ID = "big_query_storage";
+  private static final String TABLE_PREFIX = "storage_write_";
+
+  private BigQueryOptions bqOptions;
+  private static final BigqueryClient BQ_CLIENT = new BigqueryClient("BigQueryStorageIOWriteIT");
+
+  private void setUpTestEnvironment(WriteMode writeMode) {
+    PipelineOptionsFactory.register(BigQueryOptions.class);
+    bqOptions = TestPipeline.testingPipelineOptions().as(BigQueryOptions.class);
+    bqOptions.setProject(TestPipeline.testingPipelineOptions().as(GcpOptions.class).getProject());
+    if (writeMode == WriteMode.EXACT_ONCE) {
+      bqOptions.setUseStorageWriteApi(true);
+    } else {
+      bqOptions.setUseStorageWriteApiAtLeastOnce(true);
+    }
+    bqOptions.setNumStorageWriteApiStreams(2);
+    bqOptions.setStorageWriteApiTriggeringFrequencySec(1);
+    project = TestPipeline.testingPipelineOptions().as(GcpOptions.class).getProject();
+  }
+
+  static class FillRowFn extends DoFn<Long, TableRow> {
+    @ProcessElement
+    public void processElement(ProcessContext c) {
+      c.output(new TableRow().set("number", c.element()).set("str", "aaaaaaaaaa"));
+    }
+  }
+
+  private void runBigQueryIOStorageWritePipeline(int rowCount, WriteMode writeMode) {
+    String tableName = TABLE_PREFIX + System.currentTimeMillis();
+    TableSchema schema =
+        new TableSchema()
+            .setFields(
+                ImmutableList.of(
+                    new TableFieldSchema().setName("number").setType("INTEGER"),
+                    new TableFieldSchema().setName("str").setType("STRING")));
+
+    Pipeline p = Pipeline.create(bqOptions);
+    p.apply("Input", GenerateSequence.from(0).to(rowCount))
+        .apply("GenerateMessage", ParDo.of(new FillRowFn()))
+        .apply(
+            "WriteToBQ",
+            BigQueryIO.writeTableRows()
+                .to(String.format("%s:%s.%s", project, DATASET_ID, tableName))
```

Comment:
Looking at the job executions, seems like tests ran using BATCH_LOADS mode for some reason.
https://pantheon.corp.google.com/dataflow/jobs/us-central1/2022-03-09_17_22_34-16871784130096288394;step=?project=apache-beam-testing&pageState=(%22dfTime%22:(%22l%22:%22dfJobMaxTime%22))

https://pantheon.corp.google.com/dataflow/jobs/us-central1/2022-03-09_17_22_34-16871784130096288394?project=apache-beam-testing&pageState=(%22dfTime%22:(%22l%22:%22dfJobMaxTime%22))


Can you try setting following ?

"withMethod(BigQueryIO.Write.Method.STORAGE_WRITE_API)" or "withMethod(BigQueryIO.Write.Method.STORAGE_API_AT_LEAST_ONCE).